parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -1131,7 +1227,9 @@ public void writeDataPageV2(
    * @param pageHeaderAAD pageHeader AAD
    * @param sizeStatistics size statistics for the page
    * @throws IOException if any I/O error occurs during writing the file
+   * @deprecated will be removed in 2.0.0. Use {@link ParquetFileWriter#writeDataPageV2(int, int, int, BytesInput, BytesInput, Encoding, BytesInput, boolean, int, Statistics, BlockCipher.Encryptor, byte[], SizeStatistics)} instead
    */
+  @Deprecated
```

Comment:
Not related to this PR: we have plenty of deprecated public methods with a bunch of parameters like this. It takes pain for the downstream to migrate after removing them in the 2.0.0 and still likely to be deprecated if we need a new parameter next time. Should we change these into a `Builder` or a class with many parameters so we won't break anything in the future? @gszadovszky @Fokko

Reply:
Separating those methods from `ParquetFileWriter` class and reducing the duplication between `ColumnChunkPageWriteStore` could help the maintenance.